src/lib/error/bad-data-error.ts

```diff
@@ -101,32 +101,62 @@ const oneOfMessage = (
     };
 };
 
+const enumMessage = (
+    propertyName: string,
+    message: string | undefined,
+    allowedValues: string[],
+    suppliedValue: string | null | undefined,
+) => {
+    const description = `The \`${propertyName}\` property ${
+        message ?? 'must match one of the allowed values'
+    }: ${allowedValues
+        .map((value) => `"${value}"`)
+        .join(
+            ', ',
+        )}. You provided "${suppliedValue}", which is not valid. Please use one of the allowed values instead..`;
+
+    return {
+        description,
+        message: description,
+        path: propertyName,
+    };
+};
+
+// Sometimes, the error object contains a dataPath, even if it's not
+type ActualErrorObject = ErrorObject & { dataPath?: string };
```

Comment:
I think it's possible that the data that previously resided in `dataPath` is now available in `instancePath`, so I'll try to make a follow-up PR where I fix that.